bundles/pubsub/pubsub_utils/src/pubsub_serializer_handler.c

```diff
@@ -0,0 +1,360 @@
+        status = entry->svc->serialize(entry->svc->handle, input, output, outputIovLen);
+    } else {
+        status = CELIX_ILLEGAL_ARGUMENT;
+        L_ERROR("Cannot find message serialization service for msg id %u.", msgId);
+    }
+    celixThreadRwlock_unlock(&handler->lock);
+    return status;
+}
+
+celix_status_t pubsub_serializerHandler_freeSerializedMsg(pubsub_serializer_handler_t* handler, uint32_t msgId, struct iovec* input, size_t inputIovLen) {
+    celix_status_t status = CELIX_SUCCESS;
+    celixThreadRwlock_readLock(&handler->lock);
+    pubsub_serialization_service_entry_t* entry = findEntry(handler, msgId);
+    if (entry != NULL) {
+        entry->svc->freeSerializedMsg(entry->svc->handle, input, inputIovLen);
+    } else {
+        status = CELIX_ILLEGAL_ARGUMENT;
+        L_ERROR("Cannot find message serialization service for msg id %u.", msgId);
+    }
+    celixThreadRwlock_unlock(&handler->lock);
+    return status;
+
+}
+
+celix_status_t pubsub_serializerHandler_deserialize(pubsub_serializer_handler_t* handler, uint32_t msgId, int serializedMajorVersion, int serializedMinorVersion, const struct iovec* input, size_t inputIovLen, void** out) {
+    celix_status_t status;
+    celixThreadRwlock_readLock(&handler->lock);
+    pubsub_serialization_service_entry_t* entry = findEntry(handler, msgId);
+    bool compatible = false;
+    if (entry != NULL) {
+        compatible = isCompatible(handler, entry, serializedMajorVersion, serializedMinorVersion);
+        if (compatible) {
+            status = entry->svc->deserialize(entry->svc->handle, input, inputIovLen, out);
+        } else {
+            status = CELIX_ILLEGAL_ARGUMENT;
+            char *version = celix_version_toString(entry->msgVersion);
+            L_ERROR("Cannot deserialize for message %s version %s. The serialized input has a version of %d.%d.x and this is incompatible.", entry->msgFqn, version, serializedMajorVersion, serializedMinorVersion);
+            free(version);
+        }
+    } else {
+        status = CELIX_ILLEGAL_ARGUMENT;
+        L_ERROR("Cannot find message serialization service for msg id %u.", msgId);
+    }
+    celixThreadRwlock_unlock(&handler->lock);
+    return status;
+}
+
+celix_status_t pubsub_serializerHandler_freeDeserializedMsg(pubsub_serializer_handler_t* handler, uint32_t msgId, void* msg) {
+    celix_status_t status = CELIX_SUCCESS;
+    celixThreadRwlock_readLock(&handler->lock);
+    pubsub_serialization_service_entry_t* entry = findEntry(handler, msgId);
+    if (entry != NULL) {
+        entry->svc->freeDeserializedMsg(entry->svc->handle, msg);
+    } else {
+        status = CELIX_ILLEGAL_ARGUMENT;
+        L_ERROR("Cannot find message serialization service for msg id %u.", msgId);
+    }
+    celixThreadRwlock_unlock(&handler->lock);
+    return status;
+}
+
+bool pubsub_serializerHandler_supportMsg(pubsub_serializer_handler_t* handler, uint32_t msgId, int serializedMajorVersion, int serializedMinorVersion) {
```

Comment:
Does this check if a message can be (de)serialized? If so -> isMessageSupported. If not, what does it do?

Reply:
renamed and added doc